qa-tests-backend/scripts/run-compatibility.sh

```diff
@@ -20,6 +20,24 @@ source "$ROOT/qa-tests-backend/scripts/lib.sh"
 
 set -euo pipefail
 
+shorten_tag() {
+    if [[ "$#" -ne 1 ]]; then
+        die "Expected a version tag as parameter in shorten_tag: shorten_tag <tag>"
+    fi
+
+    long_tag="$1"
+
+    short_tag_regex='([0-9]+\.[0-9]+\.[0-9xX]+)'
+
+    if [[ $long_tag =~ $short_tag_regex ]]; then
+        echo "${BASH_REMATCH[1]}"
+    else
+        echo "${long_tag}"
+        >&2 echo "Failed to shorten tag ${long_tag} as it did not match the regex: \"${short_tag_regex}\""
+        exit 1
```

Comment:
Let's not kill the script. We returned the long version and produced a log line, so the script may continue. 
We wouldn't like to interrupt the CI because of such thing.
```suggestion
```

Reply:
From my understanding of bash exit 1 does not actually kill the script but just sends an exit code denoting that the function failed. We don't catch this or do anything with it, so the tests keep running as expected, but this exit 1 makes it clearer to people running the code that this is a failure state.

Let me know your thoughts on that, if I'm wrong I don't mind deleting it either.

Reply:
Have you tried with `set -e`? :)

Reply:
Yes, just tried it now. There is no issue either. I also tried `set -euo pipefail` like most of our scripts have and everything is good. But I can remove the exit 1 anyways if you think it's confusing or unnecessary